iceberg.bucket-execution=true failing to load a partitioned, bucketed table

Trino version is 470, fault tolerance enabled.

We noticed an error when running some queries on a iceberg partitioned table partitioned something like:
ARRAY['name_column', 'bucket(key_column, 32)']

Digging in, I noticed a [fairly recent commit](https://github.com/trinodb/trino/commit/3359c280ebf610b3fde8dd30958120158a8b951a) that looked promising. I disabled `iceberg.bucket-execution` via the session properties, and the query started returning results instead of erroring. The error is below:

```
org.jkiss.dbeaver.model.exec.DBCException: SQL Error [65536]: Query failed (#20250224_190940_00378_4ruzs): class java.lang.Integer cannot be cast to class java.lang.Long (java.lang.Integer and java.lang.Long are in module java.base of loader 'bootstrap')
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCResultSetImpl.nextRow(JDBCResultSetImpl.java:193)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.fetchQueryData(SQLQueryJob.java:938)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:704)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$2(SQLQueryJob.java:546)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:192)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:565)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.extractData(SQLQueryJob.java:1039)
	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.readData(SQLEditor.java:4309)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:128)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:192)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:126)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$ResultSetDataPumpJob.run(ResultSetViewer.java:5071)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:119)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.sql.SQLException: Query failed (#20250224_190940_00378_4ruzs): class java.lang.Integer cannot be cast to class java.lang.Long (java.lang.Integer and java.lang.Long are in module java.base of loader 'bootstrap')
	at io.trino.jdbc.ResultUtils.resultsException(ResultUtils.java:33)
	at io.trino.jdbc.AsyncResultIterator.lambda$new$1(AsyncResultIterator.java:93)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.Long (java.lang.Integer and java.lang.Long are in module java.base of loader 'bootstrap')
	at io.trino.plugin.iceberg.IcebergBucketFunction.applyAsInt(IcebergBucketFunction.java:88)
	at io.trino.plugin.iceberg.IcebergBucketFunction.applyAsInt(IcebergBucketFunction.java:38)
	at io.trino.sql.planner.NodePartitioningManager.lambda$getSplitToBucket$7(NodePartitioningManager.java:335)
	at io.trino.execution.scheduler.faulttolerant.FaultTolerantPartitioningScheme.getPartition(FaultTolerantPartitioningScheme.java:71)
	at io.trino.execution.scheduler.faulttolerant.EventDrivenTaskSource.getSplitPartition(EventDrivenTaskSource.java:215)
	at com.google.common.collect.CollectCollectors.lambda$toImmutableListMultimap$18(CollectCollectors.java:369)
	at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:727)
	at io.trino.execution.scheduler.faulttolerant.EventDrivenTaskSource.process(EventDrivenTaskSource.java:204)
	at com.google.common.util.concurrent.AbstractTransformFuture$TransformFuture.doTransform(AbstractTransformFuture.java:260)
	at com.google.common.util.concurrent.AbstractTransformFuture$TransformFuture.doTransform(AbstractTransformFuture.java:250)
	at com.google.common.util.concurrent.AbstractTransformFuture.run(AbstractTransformFuture.java:130)
	at io.trino.$gen.Trino_470____20250224_183753_2.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1575)

```

Potentially related, Trino asks for bucketing to be specified as bucket(x, nbuckets)

Source: [Iceberg connector — Trino 471 Documentation](https://trino.io/docs/current/connector/iceberg.html#partitioned-tables)

Iceberg expects bucketing to be specified as bucket(nbuckets, x)

Source: [DDL - Apache Iceberg™](https://iceberg.apache.org/docs/latest/spark-ddl/#partitioned-by)

If the pushdown is attempting to use iceberg metadata instead of trino's, it's possible these got reversed causing these typing issues. That could also just be coincidence, but thought I'd mention it anyway.

--- a/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergBucketFunction.java
+++ b/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergBucketFunction.java
@@ -85,7 +85,7 @@ public class IcebergBucketFunction
                 .orElseThrow(() -> new IllegalArgumentException("Split does not contain partition values"));
 
         if (singleBucketFunction) {
-            long bucket = (long) requireNonNullElse(partitionValues.getFirst(), 0);
+            long bucket = (long) requireNonNullElse(partitionValues.getFirst(), 0L);
             checkArgument(0 <= bucket && bucket < bucketCount, "Bucket value out of range: %s (bucketCount: %s)", bucket, bucketCount);
             return (int) bucket;
         }

